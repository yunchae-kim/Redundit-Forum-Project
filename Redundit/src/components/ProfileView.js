/* eslint-disable no-unused-vars */
import { useEffect, useState } from "react";
import { Button, InputGroup } from "@blueprintjs/core";
import "../style/components/ProfileView.css";
import "../style/pages/Global.css";
import userIcon from "../test_data/userIcon.png";
import { getUserById } from "../utility/axiosUser";
import httpClient from "../utility/auth.helper";

const ProfileView = ({ userId, onClick }) => {
  const [user, setUser] = useState([]);
  const [id, setId] = useState(0);

  useState(() => {
    getUserById(userId).then((userFetched) => {
      setUser(userFetched);
    });
  }, []);

  useEffect(() => {
    const userObj = httpClient.getCurrentUser();
    if (userObj) {
      const { iduser } = userObj;
      if (iduser) {
        setId(iduser);
      }
    }
  }, [id]);

  return (
    <div className="ProfileView">
      {id === userId && <Button text="Edit" onClick={onClick} />}
      <div className="PV-photo">
        <div className="PV-firstCol">Profile Photo</div>
        <img className="PV-secondCol PV-userIcon" src={user.image === "dummyURL" ? userIcon : user.image} alt="" />
      </div>
      <div className="PV-email">
        <div className="PV-firstCol">Username</div>
        <div className="PV-secondCol PV-userEmail">{user.username}</div>
      </div>
      <div className="PV-joindate">
        <div className="PV-firstCol">Join Date</div>
        <div className="PV-secondCol PV-joindate">{user.joindate}</div>
      </div>
      <div className="PV-status">
        <div className="PV-firstCol">Status</div>
        <div className="PV-secondCol PV-status">{user.status ? "Active" : "Deactivated"}</div>
      </div>
    </div>
  );
};

export default ProfileView;
