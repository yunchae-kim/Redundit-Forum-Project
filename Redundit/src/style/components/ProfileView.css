.ProfileView {
    width: 100%;
    padding: 2.5rem;
    border-radius: 20px;
    background-color: #fff;
  }

.PV-name, .PV-email, .PV-password, .PV-status, .PV-joindate {
  display: flex;
}
.PV-photo {
  display: flex;
  flex-direction: row;
  height: 100px;
}

.PV-firstCol {
  width: 150px;
  height: 36px;
  margin-top: 18px;
  margin-bottom: 18px;
  font-family: HelveticaNeue;
  font-size: 18px;
  font-weight: bold;
  font-stretch: normal;
  font-style: normal;
  line-height: normal;
  letter-spacing: normal;
  align-self: center;
  text-align: left;
  color: #111;
}

.PV-secondCol {
    width: 350px;
    height: 36px;
    margin-top: 18px;
    margin-bottom: 18px;
    margin-left: 50px;
    font-family: HelveticaNeue;
    font-size: 20px;
    font-weight: normal;
    font-stretch: normal;
    font-style: normal;
    line-height: normal;
    letter-spacing: normal;
    text-align: left;
    color: #111;
  }
  
  .PV-editButton, .PV-saveButton{
    align-self: center;
    width: 50px;
    height: 20px;
  }
  
  .PV-userNameInput, .PV-userEmailInput, .PV-userPasswordInput {
    width: 300px;
    padding: 1rem 1rem;
  }

  .PV-userIcon {
    height: 80px;
    width: 80px;
    overflow: hidden;
    margin-right: 200px;
  }