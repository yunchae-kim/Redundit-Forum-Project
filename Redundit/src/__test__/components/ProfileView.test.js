import {
  waitFor,
} from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import ProfileView from "../../components/ProfileView";

test("test should ProfileView match snapshot", async () => {
  const tree = renderer.create(
    <ProfileView />,
  ).toJSON();
  await waitFor(async () => {
    expect(tree).toMatchSnapshot();
  });
});
