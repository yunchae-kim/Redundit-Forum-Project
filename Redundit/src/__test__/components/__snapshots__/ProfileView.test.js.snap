// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test should ProfileView match snapshot 1`] = `
<div
  className="ProfileView"
>
  <div
    className="PV-photo"
  >
    <div
      className="PV-firstCol"
    >
      Profile Photo
    </div>
    <img
      alt=""
      className="PV-secondCol PV-userIcon"
    />
  </div>
  <div
    className="PV-email"
  >
    <div
      className="PV-firstCol"
    >
      Username
    </div>
    <div
      className="PV-secondCol PV-userEmail"
    />
  </div>
  <div
    className="PV-joindate"
  >
    <div
      className="PV-firstCol"
    >
      Join Date
    </div>
    <div
      className="PV-secondCol PV-joindate"
    />
  </div>
  <div
    className="PV-status"
  >
    <div
      className="PV-firstCol"
    >
      Status
    </div>
    <div
      className="PV-secondCol PV-status"
    >
      Deactivated
    </div>
  </div>
</div>
`;
